stanford_events.module

```diff
@@ -7,16 +7,234 @@
-  // just add a library to all pages but rather, conditionally require this
-  // library only where it is needed.
-  // See: https://www.drupal.org/node/2274843
+  $allow_list_term = [
+    'stanford_event_terms_utility',
+  ];
+
+  $id = $view->id();
+
+  // Node Base Table Views.
+  if (in_array($id, $allow_list_node)) {
+    $output['#attached']['library'][] = 'stanford_events/event_views';
+
+    $node_list_position = array_search('node_list', $output['#cache']['tags']);
+    unset($output['#cache']['tags'][$node_list_position]);
+    foreach ($view->filter['type']->value as $node_type) {
+      $output['#cache']['tags'][] = "node_list:$node_type";
+    }
+  }
+
+  // Term Base Table Views.
+  if (in_array($id, $allow_list_term)) {
+    $output['#attached']['library'][] = 'stanford_events/event_views';
+
+    $term_list_position = array_search('term_list', $output['#cache']['tags']);
+    unset($output['#cache']['tags'][$term_list_position]);
+    foreach ($view->filter['vid']->value as $term_type) {
+      $output['#cache']['tags'][] = "term_list:$term_type";
+    }
+  }
+
+}
+
+/**
+ * Implements hook_ENTITY_TYPE_view().
+ */
+function stanford_events_node_view(array &$build, EntityInterface $entity, EntityViewDisplayInterface $display, $view_mode) {
+
+  // Only run on stanford news items.
+  if ($entity->bundle() !== "stanford_event" || !node_is_page($entity)) {
+    return;
+  }
+
+  $rh_action = $entity->get('rh_action')->getString();
+  $rh_redirect = $entity->get('rh_redirect')->getString();
+
+  // Display a message to the user that this page would redirect to another
+  // location for other users.
+  if ($rh_action == "page_redirect" && !empty($rh_redirect)) {
+    \Drupal::messenger()
+      ->addWarning(t('This page will redirect to @url for other visitors.', ['@url' => $rh_redirect]));
+  }
+
+}
+
+/**
+ * Callback for views ds group title formatter.
+ *
+ * @param string $raw
+ *   The raw string from a views grouping column.
+ * @param object $obj
+ *   The loaded entity being rendered.
+ *
+ * @return void|string
+ *   The date formatted with stanford_long_day_date_year
+ */
+function stanford_events_schedule_format_date($raw, $obj) {
+  // Expecting a timestamp in this function.
+  if (!is_numeric($raw)) {
+    return;
+  }
+  return \Drupal::service('date.formatter')->format($raw, 'stanford_long_day_date_year');
+}
+
+/**
+ * Implements hook_cron().
+ */
+function stanford_events_cron() {
+
+  // Find all events that have passed since last cron run date and now.
+  // Clear cache for those particular events and clear out the views cache.
+  $last_cron = \Drupal::state()->get('system.cron_last');
+  $now = time();
+  $query = \Drupal::entityQuery('node');
+
+  // Conditions.
+  $query->condition('type', 'stanford_event');
+  $query->condition('status', 1);
+  $query->condition('su_event_date_time.end_value', $last_cron, ">=");
+  $query->condition('su_event_date_time.end_value', $now, "<=");
```

Comment:
YOu have the same conditions here. it'll clear caches for all events.

Reply:
Eh? This query is to find all events whose end date is between the last cron and now. That should be a small subset of events.

Reply:
oh i wasn't paying attention to the second parameters. ignore me